benches/benches/bevy_reflect/function.rs

```diff
@@ -0,0 +1,62 @@
+use bevy_reflect::func::{ArgList, IntoClosure, TypedFunction};
+use bevy_reflect::prelude::*;
+use criterion::{black_box, criterion_group, criterion_main, BatchSize, Criterion};
+
+criterion_group!(benches, typed, into, call, clone);
+criterion_main!(benches);
+
+fn add(a: i32, b: i32) -> i32 {
+    a + b
+}
+
+fn typed(c: &mut Criterion) {
+    c.benchmark_group("typed")
+        .bench_function("function", |b| {
+            b.iter(|| black_box(add.get_function_info()));
```

Comment:
`Bencher::iter()` wraps the returned value of the value is `black_box()` already, so you don't need to do it! ([Source](https://docs.rs/criterion/latest/src/criterion/bencher.rs.html#88))

It may be worth swapping this and other calls with `black_box(add).get_function_info()`, so the compiler can't make assumptions about the `add()` function.

```suggestion
            b.iter(|| black_box(add).get_function_info());
```

Reply:
Oh neat! I'll change these usages

Reply:
I tried wrapping just the function in `black_box` but it resulted in a speedup in one of the benches. So I ended up not doing that